zebra/zebra_nb_state.c

```diff
@@ -89,6 +89,9 @@ lib_interface_zebra_state_zif_type_get_elem(struct nb_cb_get_elem_args *args)
 	case ZEBRA_IF_GRE:
 		type = "frr-zebra:zif-gre";
 		break;
+	case ZEBRA_IF_GRETAP:
```

Comment:
Why no changes in zebra/zapi_msg.c.  Shouldn't we just change the IS_ZEBRA_IF_GRE to encompass a GRETAP interface?

Reply:
There is a change in zebra/zapi_msg.c... or I did not understand the question.
```
--- a/zebra/zapi_msg.c
+++ b/zebra/zapi_msg.c
@@ -77,6 +77,7 @@ static void zserv_encode_interface(struct stream *s, struct interface *ifp)
        stream_putl(s, ifp->mtu6);
        stream_putl(s, ifp->bandwidth);
        stream_putl(s, zif->link_ifindex);
+       stream_putl(s, ifp->zif_type);
```

I agree that GRE and GRETAP interfaces are similar.
But what about users that already have GRETAP interfaces used today for an other purpose?

Reply:
ha ok. yes, I need to clarify. today, only NHRP uses it, so NHRP is only GRE. But we have to take care of remaining code. Later GRE_SOURCE_SET and GRE_GET zapi messages should cover other kind of tunnels too.
So I added a comment.

Reply:
I am not a big fan of XXX todo messages in the code base in all honesty.  I also do not understand your response.  Perhaps we are talking past each other a bit here or you have knowledge that I do not have.  Why do we need a IS_ZEBRA_GRE and a IS_ZEBRA_GRETAP?  What is the fundamental difference between the two?  Why do we need to treat them differently?  It's not clear to me at all.